src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -582,6 +598,33 @@ def generate_request_payload(configuration, location, public_key, tags, kubernet
         distribution=kubernetes_distro,
         infrastructure=kubernetes_infra
     )
+
+    if enable_private_link:
+        private_link_state = "Enabled" if enable_private_link.lower() == "true" else "Disabled"
+        cc = ConnectedClusterPreview(
+            location=location,
+            identity=identity,
+            agent_public_key_certificate=public_key,
+            tags=tags,
+            distribution=kubernetes_distro,
+            infrastructure=kubernetes_infra,
+            private_link_scope_resource_id=private_link_scope_resource_id,
+            private_link_state=private_link_state
+        )
+    return cc
+
+
+def generate_patch_payload(tags, enable_private_link, private_link_scope_resource_id):
+    cc = ConnectedClusterPatch(
+        tags=tags
+    )
+    if enable_private_link:
```

Comment:
It seems that we use preview cc only if `enable_private_link` is true. Should it be created when some one disable private link also?

Reply:
Yes, we use the preview SDK for both enabling & disabling private link. The if condition returns true even if the parameter '--enable-private-link' is passed as 'false'